File: LeetCode141.java
public class Solution {
    public boolean hasCycle(ListNode head) {
        Set<ListNode> node = new HashSet<ListNode>();
        while (head != null) {
            if (node.contains(head)) {
                return true;
            } else {
                node.add(head);
            }
            head = head.next;
        }
        return false;
       
    }
}

File: LeetCode141.py
class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        p1 = p2 = head
        if head:
            while p2.next and p2.next.next:
                p1 = p1.next
                p2 = p2.next.next
                if p1 == p2:
                    return True
        return False